pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -49,9 +53,31 @@ func (rs *RevisionSpec) Validate() *apis.FieldError {
 	} else if err := ValidateContainerConcurrency(rs.ContainerConcurrency, rs.ConcurrencyModel); err != nil {
 		errs = errs.Also(err)
 	}
+
+	if err := validateTimeoutSeconds(rs.TimeoutSeconds); err != nil {
+		errs = errs.Also(err)
+	}
 	return errs
 }
 
+func validateTimeoutSeconds(timeoutSeconds *metav1.Duration) *apis.FieldError {
+	if timeoutSeconds != nil {
+		if timeoutSeconds.Duration > networkingv1alpha1.MaxTimeout {
+			return apis.ErrInvalidValue(
+				fmt.Sprintf("TimeoutSeconds %v is greated than ClusterIngress MaxTimeout %v", timeoutSeconds.Duration, networkingv1alpha1.MaxTimeout),
```

Comment:
This error should be less oriented towards a contributor and more oriented towards a user.  Also I believe the first parameter here is the bad value.

Perhaps we need a new `apis.ErrOutOfBoundsValue(value, lower, upper, field)`?